### PR TITLE
Fix string formatting

### PR DESCRIPTION
**Pull request by krlmlr**
Quotes are shown in fewer cases, but always if the string contains a space.

### CODE REVIEW

File: tests/testthat/out/tibble-space.txt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+  a    
+  <chr>
+1
```
Comment (krlmlr) [inline tests/testthat/out/tibble-space.txt]: @hadley: I was thinking about showing a subtle `""` here for the empty string.?

Comment (hadley) [reply tests/testthat/out/tibble-space.txt]: Yeah, good idea

Comment (hadley) [reply tests/testthat/out/tibble-space.txt]: And maybe include `'""'` in the tests for comparison?

Comment (hadley) [reply tests/testthat/out/tibble-space.txt]: Maybe the outer quotes should get `style_subtle()`?

**Pull request merged**